src/main/java/com/google/sps/utility/GmailUtility.java

```diff
@@ -0,0 +1,155 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.utility;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.model.Message;
+import com.google.sps.model.MessageFormat;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Contains logic that handles GET & POST requests to the Gmail API and transforms those responses
+ * into easily usable Java types
+ */
+public class GmailUtility {
+  // Make constructor private so no instances of this class can be made
+  private GmailUtility() {}
```

Comment:
Nit: You don't necessarily need to define an empty constructor. One is automatically generated if you don't define it.

Reply:
I want to ensure that no one instantiates this class directly, so an explicitly defined empty constructor is required

Reply:
I see